### PR TITLE
[do not merge] AzureCP PR #2 - ASM VM Migration Commands

### PR DESCRIPTION
**Pull request by huangpf**
The content to be added to Changelog is as follows:
* ASM VM Migration Commands

### CODE REVIEW

File: lib/commands/asm/migration/migration._js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+exports.init = function (cli) {
+
+/*
+  Deployment AbortMigration
+  --service-name
+  --deployment-name
+*/
+  var deploymentAbortMigration = cli.category('invoke').description('Commands to invoke service management operations.').category('deployment')
+  .description($('Commands to manage your deployment.  '));
+  deploymentAbortMigration.command('abort-migration [service-name] [deployment-name]')
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: Are we still doing this or changing the format as discussed?

File: lib/commands/asm/migration/migration._js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    if (!deploymentName) {
+      deploymentName = cli.interaction.promptIfNotGiven($('deployment-name : '), deploymentName, _);
+    }
+
+    cli.output.verbose('deploymentName = ' + deploymentName);
+    var parametersObj = null;
+    if (options.parameterFile) {
+      cli.output.verbose('Reading file content from: \"' + options.parameterFile + '\"');
+      var parametersFileContent = fs.readFileSync(options.parameterFile, 'utf8');
+      parametersObj = JSON.parse(parametersFileContent);
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: try catch please

File: lib/commands/asm/migration/migration._js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      parametersObj.subNetName = subNetName;
+      cli.output.verbose('virtualNetworkName = ' + virtualNetworkName);
+      parametersObj.virtualNetworkName = virtualNetworkName;
+    }
+    cli.output.verbose('parametersObj = ' + JSON.stringify(parametersObj));
+    var subscription = profile.current.getSubscription(options.subscription);
+    var computeManagementClient = utils.createComputeClient(subscription);
+    var result = computeManagementClient.deployments.prepareMigration(serviceName, deploymentName, parametersObj, _);
+    if (result) {
+      cli.output.json(result);
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: are we only displaying the result in json format? No tabular/normal display?

File: lib/commands/asm/migration/migration._js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    cli.output.verbose('serviceName = ' + serviceName);
+    if (!deploymentName) {
+      deploymentName = cli.interaction.promptIfNotGiven($('deployment-name : '), deploymentName, _);
+    }
+
+    cli.output.verbose('deploymentName = ' + deploymentName);
+    var subscription = profile.current.getSubscription(options.subscription);
+    var computeManagementClient = utils.createComputeClient(subscription);
+    var result = computeManagementClient.deployments.commitMigration(serviceName, deploymentName, _);
+    if (result) {
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: only json??

File: lib/commands/asm/migration/migration._js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+      cli.output.json(result);
+    }
+  });
+/*
+  Deployment CommitMigration
+  --service-name
+  --deployment-name
+*/
+  var deploymentCommitMigration = cli.category('invoke').description('Commands to invoke service management operations.').category('deployment')
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: I hope we don't have the invoke category. It makes no sense to me.
`azure service deployment-migration prepare|commit|abort`

File: lib/commands/asm/migration/migration._js (hunk 1 above)
Comment (huangpf) [inline lib/commands/asm/migration/migration._js]: It's not finalized yet. Let me close it for now and polish it... :-)

File: lib/commands/asm/migration/migration._js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+  .description($('Commands to manage your deployment.  '));
+  deploymentCommitMigration.command('commit-migration [service-name] [deployment-name]')
+  .description($('The Commit Deployment Operation validates and commits your deployment for IaaS Classic to ARM migration.'))
+  .usage('[options] <service-name> <deployment-name>')
+  .option('--service-name <service-name>', $('service-name'))
+  .option('--deployment-name <deployment-name>', $('deployment-name'))
```
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: -n --name  "The deployment name"

File: lib/commands/asm/migration/migration._js (hunk 1 above)
Comment (amarzavery) [inline lib/commands/asm/migration/migration._js]: Yup no worries. Just commenting, so you can verify that the list of things have been fixed.

**Pull request closed without merging**